src/Symfony/Component/Routing/Loader/LoaderHelper.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+/**
+ * Common parsing logic between route loaders.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+trait LoaderHelper
+{
+    private function redirectToDefaults(string $redirect, bool $permanent = false): array
+    {
+        if ($this->looksLikeUrl($redirect)) {
+            $defaults['_controller'] = 'Symfony\Bundle\FrameworkBundle\Controller\RedirectController::urlRedirectAction';
+            $defaults['path'] = $redirect;
+        } else {
+            $defaults['_controller'] = 'Symfony\Bundle\FrameworkBundle\Controller\RedirectController::redirectAction';
+            $defaults['route'] = $redirect;
+        }
+
+        if ($permanent) {
+            $defaults['permanent'] = true;
+        }
+
+        return $defaults;
+    }
+
+    private function looksLikeUrl(string $urlOrRouteName): bool
+    {
+        foreach (array('/', '//', 'http://', 'https://') as $pattern) {
+            if (substr($urlOrRouteName, 0, strlen($pattern)) == $pattern) {
```

Comment:
should use `0 === strpos()` instead to be consistent with the codebase (we avoid allocating a new string)